src/panels/lovelace/cards/hui-thermostat-card.ts

```diff
@@ -107,18 +113,18 @@ export class HuiThermostatCard extends hassLocalizeLitMixin(LitElement)
     `;
   }
 
-  protected shouldUpdate(changedProps) {
+  protected shouldUpdate(changedProps: PropertyValues): boolean {
     if (changedProps.get("hass")) {
       return (
-        changedProps.get("hass").states[this.config!.entity] !==
-        this.hass!.states[this.config!.entity]
+        (changedProps.get("hass") as any).states[this._config!.entity] !==
+        this.hass!.states[this._config!.entity]
       );
     }
-    return changedProps;
+    return (changedProps as unknown) as boolean;
```

Comment:
You are typecasting things until TypeScript kept quiet? 🙄 

Just return `true` here given that there are changed props (or else this method wouldn't be called)

Reply:
Ive done this in the last few cards....

Reply:
Time to fix it 😉